Add Homing missile

weishui opened the pull request:
Add HomingMissile MonoBehaviour
Add Big Fireball projectile
Change the tag of PlayerShip to "Player"

weishui committed:
Add Homing missile
Add HomingMissile MonoBehaviour
Add Big Fireball projectile
Change the tag of PlayerShip to "Player"

weishui committed:
ReUpload

jimraynor0 commented:
我试验了一下有两个奇怪的地方
1. 第一批导弹刚发射的时候都是向上飞的
2. 有些子弹在边界直接就消失了
看看有没有办法修一下

另外看起来最好现在就把子弹的速度和飞行时候的逻辑挪到子弹的script里面。turret只需要按比例instantiate子弹就行了。我建议每类子弹有一个pool

weishui committed:
BugFix
Add OnEnable() in HomingMissile to set initial transform.rotation
Switch places of projectile transform setup & setactive in RandomTurrets
.Fire(), to fix projectiles mistakenly initialized to wrong direction

weishui commented:
> 我试验了一下有两个奇怪的地方
> 
> 1. 第一批导弹刚发射的时候都是向上飞的
> 2. 有些子弹在边界直接就消失了
>    看看有没有办法修一下
> 
> 另外看起来最好现在就把子弹的速度和飞行时候的逻辑挪到子弹的script里面。turret只需要按比例instantiate子弹就行了。我建议每类子弹有一个pool

做了最小的改动fix了两个问题，一个是hommingMissile增加了OnEnable()来设置初始Rotation，一个是RandomTurret.Fire()的SetActive放在了transform和rb设置之后。
